### PR TITLE
Chao9441/upgrade to earth v1.11

### PR DESCRIPTION
**Pull request by gis2all**

### CODE REVIEW

File: src/ArcGISEarth.AutoAPI.Utils/AutomationAPIHelper.cs [hunk 1]
```diff
@@ -0,0 +1,268 @@
+﻿// Copyright 2018 Esri
```
Comment (yong7743) [inline src/ArcGISEarth.AutoAPI.Utils/AutomationAPIHelper.cs:1]: should be 2020

File: src/ArcGISEarth.AutoAPI.Utils/ArcGISEarth.AutoAPI.Utils.csproj [hunk 2]
```diff
@@ -43,14 +44,15 @@
...
+    <Compile Include="AutomationAPIHelper.cs" />
     <Compile Include="Properties\AssemblyInfo.cs" />
   </ItemGroup>
   <ItemGroup>
     <PackageReference Include="Newtonsoft.Json">
-      <Version>11.0.2</Version>
+      <Version>12.0.3</Version>
+    </PackageReference>
+    <PackageReference Include="RestSharp">
+      <Version>106.11.5-alpha.0.2</Version>
```
Comment (yong7743) [inline src/ArcGISEarth.AutoAPI.Utils/ArcGISEarth.AutoAPI.Utils.csproj:55]: The New 3rd party library needs to be reviewed by License team.  And you'd better use a normal release version, not a an alpha version.

**Comment by bentan2013** [comment]
> Should we consider to migrate the test project to .NET Core?

All is well. Depend on @gis2all

File: src/ArcGISEarth.AutoAPI.Utils/AutomationAPIHelper.cs [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+        }
+
+        public string GetWorkspace()
+        {
+            try
+            {
+                var request = new RestRequest(_workspaceRequestUrl, Method.GET);
+                request.AddParameter("undefind", _workspaceRequestUrl);
+                request.AddHeader("accept", "*/*");
+                request.AddHeader("Content-Type", "image/jpeg");
```
Comment (bentan2013) [inline src/ArcGISEarth.AutoAPI.Utils/AutomationAPIHelper.cs:226]: not request jpeg

File: src/ArcGISEarth.AutoAPI.Utils/AutomationAPIHelper.cs [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+using Newtonsoft.Json.Linq;
+using RestSharp;
+using System;
+using System.IO;
+
+namespace ArcGISEarth.AutoAPI.Utils
+{
+    public class AutomationAPIHelper
+    {
+        private const string TARGET_OPERATIONALLAYERS = "OperationalLayers";
```
Comment (bentan2013) [inline src/ArcGISEarth.AutoAPI.Utils/AutomationAPIHelper.cs:23]: use this `operationalLayers`

File: src/ArcGISEarth.AutoAPI.Utils/AutomationAPIHelper.cs [hunk 5]
```diff
@@ -0,0 +1,268 @@
...
+
+        private const string TARGET_BASEMAPS = "BasemapLayers";
```
Comment (bentan2013) [inline src/ArcGISEarth.AutoAPI.Utils/AutomationAPIHelper.cs:25]: use this `baseMaps`

File: src/ArcGISEarth.AutoAPI.Utils/AutomationAPIHelper.cs (hunk 4 above)
Comment (bentan2013) [inline src/ArcGISEarth.AutoAPI.Utils/AutomationAPIHelper.cs:23]: use this `operationalLayers`

File: src/ArcGISEarth.AutoAPI.Utils/AutomationAPIHelper.cs [hunk 6]
```diff
@@ -0,0 +1,268 @@
...
+
+        private const string TARGET_ELEVATIONLAYERS = "ElevationLayers";
```
Comment (bentan2013) [inline src/ArcGISEarth.AutoAPI.Utils/AutomationAPIHelper.cs:27]: use this `elevationLayers`

File: src/ArcGISEarth.AutoAPI.Utils/AutomationAPIHelper.cs [hunk 7]
```diff
@@ -0,0 +1,268 @@
...
+        }
+
+        public string ClearLayers(string inputJsonStr)
+        {
+            try
+            {
+                JObject jobject = JObject.Parse(inputJsonStr);
+                string tartget = jobject["target"].ToString();
+                RestRequest request = null;
+                if (tartget.Equals(TARGET_OPERATIONALLAYERS, StringComparison.OrdinalIgnoreCase))
```
Comment (bentan2013) [inline src/ArcGISEarth.AutoAPI.Utils/AutomationAPIHelper.cs:161]: just use the value in docs. 

use the same input logic as remove layer. e.g. just `baseMaps`
```
        public string RemoveLayer(string layerId)
        {
            try
            {
                string layerIdUrl = $"{_layerRequestUrl}/{layerId}";
```
Sorry for this
你问我为什么ClearLayers的方式不和layer一样时，我以为你说的是上一个版本，上一个版本1.8有可能多个key。这个版本是一致的。

File: src/ArcGISEarth.AutoAPI.Utils/AutomationAPIHelper.cs [hunk 8]
```diff
@@ -0,0 +1,268 @@
...
+        public string GetSnapshot(string imagePath)
+        {
+            try
+            {
+                var request = new RestRequest(_snapshotRequestUrl, Method.GET);
+                request.AddParameter("undefind", _snapshotRequestUrl);
+                request.AddHeader("accept", "*/*");
+                request.AddHeader("Content-Type", "image/jpeg");
+                IRestResponse response = _client.Execute(request);
+                Uri uri = new Uri(imagePath);
```
Comment (bentan2013) [inline src/ArcGISEarth.AutoAPI.Utils/AutomationAPIHelper.cs:197]: consider `File.Exists(imagePath)`

File: src/ArcGISEarth.AutoAPI.Utils/AutomationAPIHelper.cs (hunk 3 above)
Comment (bentan2013) [inline src/ArcGISEarth.AutoAPI.Utils/AutomationAPIHelper.cs:226]: not need

**Comment by gis2all** [comment]
@yong7743  Have updated all except migrating to .Net Core

1. Replace RESTClient with HttpClinet

image

2. Custom api url address

image

File: src/ArcGISEarth.AutoAPI.Utils/AutomationAPIHelper.cs [hunk 9]
```diff
@@ -0,0 +1,286 @@
...
+                    return "Save snapshot Failed, Please type correct file path!";
+                }
+            }
+            catch (Exception ex)
+            {
+                return ex.Message;
+            }
+        }
+
+        private HttpContent ConverteToHttpContent(string str)
```
Comment (yong7743) [inline src/ArcGISEarth.AutoAPI.Utils/AutomationAPIHelper.cs]: It should be Json, so the name should be ConvertJsonToHttpContent(string json)

**Pull request closed without merging**